File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/legacy/rest/model/login/RegistrationParams.java
/*
 * Copyright 2014 OpenMarket Ltd
 * Copyright 2017 Vector Creations Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package im.vector.matrix.android.internal.legacy.rest.model.login;

import java.util.Map;

/**
 * Class to pass parameters to the different registration types for /register.
 */
public class RegistrationParams {
    // authentification parameters
    public Map<String, Object> auth;

    // the account username
    public String username;

    // the account password
    public String password;

    // With email
    public Boolean bind_email;

    // With phone_number
    public Boolean bind_msisdn;

    // device name
    public String initial_device_display_name;

    // Temporary flag to notify the server that we support msisdn flow. Used to prevent old app
    // versions to end up in fallback because the HS returns the msisdn flow which they don't support
    public Boolean x_show_msisdn;
}

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/registration/RegistrationParams.kt
/*
 * Copyright 2025 New Vector Ltd.
 * Copyright 2020 The Matrix.org Foundation C.I.C.
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */
package org.matrix.android.sdk.internal.auth.registration

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Class to pass parameters to the different registration types for /register.
 */
@JsonClass(generateAdapter = true)
internal data class RegistrationParams(
        // authentication parameters
        @Json(name = "auth")
        val auth: AuthParams? = null,

        // the account username
        @Json(name = "username")
        val username: String? = null,

        // the account password
        @Json(name = "password")
        val password: String? = null,

        // device name
        @Json(name = "initial_device_display_name")
        val initialDeviceDisplayName: String? = null,

        // Temporary flag to notify the server that we support msisdn flow. Used to prevent old app
        // versions to end up in fallback because the HS returns the msisdn flow which they don't support
        @Json(name = "x_show_msisdn")
        val xShowMsisdn: Boolean? = null
)
